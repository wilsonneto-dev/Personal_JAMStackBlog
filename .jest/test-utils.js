import { render } from '@testing-library/react';

import { ThemeProvider } from 'styled-components';
import defaultTheme from 'styles/themes/default';

const WithTheme = ({ children }) => {
  return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>;
};

export const renderWithTheme = (ui, options) =>
  render(ui, { wrapper: WithTheme, ...options });
