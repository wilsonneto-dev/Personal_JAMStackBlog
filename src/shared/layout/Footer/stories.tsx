import { Story, Meta } from '@storybook/react/types-6-0';
import Footer from '.';

export default {
  title: 'Layout/Footer',
  component: Footer
} as Meta;

export const Default: Story = () => <Footer />;
