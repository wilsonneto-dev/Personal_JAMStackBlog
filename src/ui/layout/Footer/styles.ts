import styled from 'styled-components';

export const Wrapper = styled.footer`
  background-color: ${(props) => props.theme.colors.featuredBackground};
`;

export const InnerWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: min(${(props) => props.theme.sizes.box}, 100%);
  padding: 6rem 0 8rem;
  margin: 0 auto;
  height: 100%;
`;

export const Logo = styled.img`
  height: 4rem;
`;

export const SocialLinks = styled.div`
  margin-top: 1rem;
`;

export const SocialIcon = styled.a`
  display: inline-block;
  img {
    height: 1.6rem;
    margin-left: 0.5rem;
  }
`;
