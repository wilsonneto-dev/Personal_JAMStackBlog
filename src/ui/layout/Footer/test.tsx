import { screen } from '@testing-library/react';
import { renderWithTheme } from '../../../../.jest/test-utils';

import Footer from '.';

describe('<Footer />', () => {
  it('should render the logo', () => {
    renderWithTheme(<Footer />);
    expect(screen.getByRole('logo')).toBeInTheDocument();
  });

  it('should render social links', () => {
    renderWithTheme(<Footer />);
    expect(screen.getAllByRole('social-link').length).toBeGreaterThan(0);
  });

  it('should looks like the snapshot', () => {
    const { container } = renderWithTheme(<Footer />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
