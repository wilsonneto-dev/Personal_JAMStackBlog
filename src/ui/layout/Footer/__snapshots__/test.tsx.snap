// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should looks like the snapshot 1`] = `
.c0 {
  background-color: #612F74;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: min(100em,100%);
  padding: 6rem 0 8rem;
  margin: 0 auto;
  height: 100%;
}

.c2 {
  height: 4rem;
}

.c3 {
  margin-top: 1rem;
}

.c4 {
  display: inline-block;
}

.c4 img {
  height: 1.6rem;
  margin-left: 0.5rem;
}

<footer
  class="c0"
>
  <div
    class="c1"
  >
    <div>
      <a
        href="/"
      >
        <img
          alt="Logo Wilson Neto"
          class="c2"
          role="logo"
          src="/img/wilsonneto.png"
        />
      </a>
    </div>
    <div
      class="c3"
    >
      <a
        class="c4"
        href="/"
        role="social-link"
        target="_blank"
      >
        <img
          alt="Wilson Neto - Github"
          src="/img/ic-github.svg"
        />
      </a>
      <a
        class="c4"
        href="/"
        role="social-link"
        target="_blank"
      >
        <img
          alt="Wilson Neto - Linkedin"
          src="/img/ic-linkedin.svg"
        />
      </a>
    </div>
  </div>
</footer>
`;
